Simulator/player.py

```diff
@@ -326,34 +327,41 @@ def generate_champion_vectors(self):
     def generate_single_champion_vector(self, curr_champ, champion_info_array):
         '''
         Helps to generate a vector of length CHAMPION_INFO
+        champion number (1 spot), champion star level(1 spot), champion cost (1 spot),
+        chosen (1 spot) , past combat (1 spot), 3 items (6 spot)
+
         :param curr_champ: Object champion
         :param champion_info_array: The array to update the information in
         :return:
         '''
+
         # start with champion name
         c_index = list(COST.keys()).index(
             curr_champ.name) + 1  # Returns index of champion, # Avoiding index 0 as index 0 is reserved for no chammpion
         # This should update the champion name section of the vector
         champion_info_array[0] = float(c_index) / self.MAX_CHAMPION_IN_SET
-        champion_info_array[1] = curr_champ.stars
-        champion_info_array[2] = curr_champ.cost
+        champion_info_array[1] = curr_champ.stars / 3
+        champion_info_array[2] = curr_champ.cost / 5
 
         if curr_champ.chosen:
             champion_info_array[3] = 1
-
         if curr_champ.survive_combat:
             champion_info_array[4] = 1
         elif curr_champ.participated_in_combat:  # Did not survive combat
-            champion_info_array[5] = 0.5
+            champion_info_array[4] = 0.5
         # else 0 , 0 implies that did not participate in combat
 
         item_arr = np.zeros(6)
-        for index, item in enumerate(curr_champ.items):
-            item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
-            # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now
-            # first spot of each of the item is left empty for now
-            item_arr[index * 2] = float(item_index) / self.MAX_ITEMS_IN_SET
-        champion_info_array[6:] = item_arr
+        for ind, item in enumerate(curr_champ.items):
+            if ind <= 2:
+                item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now

```

Comment:
Look at the item_stats.py --> item_builds. You can use that dictionary of lists to get the components that you are looking for.